packages/core/src/components/overlay/overlay.tsx

```diff
@@ -314,17 +314,19 @@ export class Overlay extends React.PureComponent<IOverlayProps, IOverlayState> {
         }
         openStack.push(this);
 
-        if (this.props.canOutsideClickClose && !this.props.hasBackdrop) {
-            document.addEventListener("mousedown", this.handleDocumentClick);
+        if (this.props.autoFocus) {
+            this.bringFocusInsideOverlay();
```

Comment:
this should happen when not inline too, so I moved it to a more logical order in the method.